### DISCUSSION TITLE
Sendmail and tasmota32 problems

### DISCUSSION THREAD

**Discussion by marklaz** [arendst/Tasmota#22445] [Q&A]
Hello,

I tried to compile tasmota32 with integrated sendmail functionality using gitpod.

Compilation of plain tasmota32 within gitpod worked fine (Tasmota32 Version is 14.3.0.4(tasmota32)-3_1_0(2024-11-08T06:51:19)). FW works like expected on an ESP32 . 
When I activate sendmail in file user_config_override.h, compile also runs through without errors, but after flashing the FW onto the ESP32, the board seems to be reset when using the command sendmail. (The sendmail is triggered in berry code from GPIO 34 going low. Putting GPIO 34 on low works as expected, but after that the board seems to reboot each time I do so.)

I added the following lines to user_config_override.h:
#ifndef USE_SENDMAIL
#define USE_SENDMAIL
#endif

Including the following lines in the file also does not seem to help:
#ifndef USE_ESP32MAIL
#define USE_ESP32MAIL
#endif

Are there probably any additional modules I should add? (Sorry, if I overlooked something in the docs or other discussions.)
Is the any precompiled binary version of tasmota32 which already includes sendmail?

Any help is highly appreciated, thanks.

Some information on the ESP32 board:
![esp32_board](https://github.com/user-attachments/assets/e35238f8-a3ac-4d92-9810-0cb9f71cc575)

**Comment by Jason2866** [comment]
Maybe the Berry code is to blame?

**Comment by marklaz** [comment]
I did some further tests without berry, only using the console.

Command 
sendmail [smtp.xxx.yy:465:xxxx@yyy.zz:pwpwpwpwpwpwpw:xxxx@yyy.zz.de:xxxx@yyy.zz:] "Ausfall"

gives (expected):
16:07:52.802 CMD: sendmail [smtp.xxx.yy:465:xxxx@yyy.zz:pwpwpwpwpwpwpw:xxxx@yyy.zz.de:xxxx@yyy.zz:] "Ausfall"
16:07:52.806 RSL: RESULT = {"Sendmail":"Wrong parameters"}

as the subject is missing.

But issuing the command (i.e. the same as above with added subject)
sendmail [smtp.xxx.yy:465:xxxx@yyy.zz:pwpwpwpwpwpwpw:xxxx@yyy.zz.de:xxxx@yyy.zz:Tasmota] "Ausfall"

restarts the system, giving
00:00:03.997 RSL: INFO3 = {"Info3":{"RestartReason":{"Exception":1,"Reason":"Syscall","EPC":"40091c80","EXCVADDR":"00000000","CallChain":["40091c7d","40087653","40170f97","40175192","40178d0d","4016ee73","401530af","40153157","40153936","40153ac4","40153b26","40153b5e","4014d574","4014ee8c","4017b19a","400e4d8c","400ea9b3","400d44f2","400ef7ed","401347a1","400d904a","400d918a","400ecc68","40138ae1","401400f9","4014011b","401449c3","400ef800","401347a1","400dedf8","400d934c","4016ef96"]},"BootCount":153}}

Is that of any help in search for the reason, why sendmail fails?

Could I provide any additional data?

Thanks for your advice.
